core/src/dunit/scala/io/snappydata/cluster/SplitClusterDUnitSecurityTest.scala

```diff
@@ -874,6 +875,226 @@ class SplitClusterDUnitSecurityTest(s: String)
+      }
+    }).foreach(x => println(s"AFTER DROP  [snappy-jars]: ${x.getAbsolutePath}"))
+    server2Dir.listFiles(new FileFilter {
+      override def accept(pathname: File): Boolean = {
+        pathname.getName.contains("myudf") && pathname.getName.contains("jar")
+      }
+    }).foreach(x => println(s"AFTER DROP  [snappy-jars]: ${x.getAbsolutePath}"))
+
+    // Verify list jars
+    stmt2.execute(s"list jars")
+    rs = stmt2.getResultSet
+    rows = 0
+    if (rs ne null) {
+      while (rs.next()) {
+        rows += 1
+        val udfName = rs.getString(1)
+        assert(udfName.equalsIgnoreCase(s"[UDF]$jdbcUser1.myUDF"),
+          s"Unexpected UDF name $udfName")
+      }
+    }
+
+    logInfo((snappyProductDir + "/sbin/snappy-stop-all.sh").!!)
+
+    logInfo((snappyProductDir + "/sbin/snappy-start-all.sh").!!)
+
+    user1Conn = getConn(jdbcUser1)
+    // Select with the existing UDF
+    stmt2 = user1Conn.createStatement()
+    stmt2.execute(s"select myUDF('abcd')")
+    rs = stmt2.getResultSet
+    if (rs ne null) {
+      while (rs.next()) {
+        assert(rs.getInt(1) == -996, s"Expected -996 post reboot, found ${rs.getInt(1)}")
+      }
+      rs.close()
+    }
+
+    user2Conn = getConn(jdbcUser2)
+    // Select with the dropped UDF
+    stmt2 = user2Conn.createStatement()
+    try {
+      stmt2.execute(s"select myUDF('abcd')")
+      assert(false, s"Expected an exception!")
+    } catch {
+      case _: SQLException => // expected
+      case t: Throwable => assert(false, s"Unexpected exception $t")
+    } finally {
+      if (stmt2.getResultSet ne null) stmt2.getResultSet.close()
+    }
+
+    // Verify list jars
+    stmt2.execute(s"list jars")
+    rs = stmt2.getResultSet
+    rows = 0
+    if (rs ne null) {
+      while (rs.next()) {
+        rows += 1
+        val udfName = rs.getString(1)
+        assert(udfName.equalsIgnoreCase(s"[UDF]$jdbcUser1.myUDF"),
+          s"Unexpected UDF name $udfName")
+      }
+    }
+
+    val udf1SourceNew = "public class StringLengthUDF1 implements " +
+        "org.apache.spark.sql.api.java.UDF1<String,Integer> {" +
+        " @Override public Integer call(String s){ " +
+        "               return (s.length() * 1000); " +
+        "} }"
+    val file1New = SparkUtilsAccess.createUDFClass("StringLengthUDF1", udf1SourceNew)
+    val jar1New = SparkUtilsAccess.createJarFile(Seq(file1New), "user1udf", true)
+
+    stmt1 = user1Conn.createStatement()
+    executeSQL(stmt1, s"DROP FUNCTION myUDF")
+
+    stmt2 = user1Conn.createStatement()
+    try {
+      stmt2.execute(s"select myUDF('abcd')")
+      assert(false, s"Expected an exception!")
+    } catch {
+      case _: SQLException => // expected
+      case t: Throwable => assert(false, s"Unexpected exception $t")
+    }
+
+    // Verify list jars
+    stmt2.execute(s"list jars")
+    rs = stmt2.getResultSet
+    if (rs ne null) {
+      while (rs.next()) {
```

Comment:
you can assert just rs.next() == false.

rs.next() == true is an illegal state.

Reply:
Yes, that would be cleaner, will do. BTW, I'm asserting FALSE right after `rs.next()` which ensures that `rs.next() == true` is invalid.